src/content/platform-integration/platform-channels.md

```diff
@@ -6,98 +6,86 @@ description: Learn how to write custom platform-specific code in your app.
 
 <?code-excerpt path-base="platform_integration"?>
 
-This guide describes how to write custom platform-specific code.
-Some platform-specific functionality is available
-through existing packages;
-see [using packages][].
+This guide describes how to use custom platform-specific
+code with Flutter.
 
-[using packages]: /packages-and-plugins/using-packages
+## Overview
 
-:::note
-The information in this page is valid for most platforms,
-but platform-specific code for the web generally uses
-[JS interoperability][] or the [`dart:html` library][] instead.
-:::
+You can use platform-specific code in your Flutter app. 
+A few common ways to do this include:
 
-Flutter uses a flexible system that allows you to call
-platform-specific APIs in a language that works directly
-with those APIs:
+* Use Flutter's platform channel APIs to pass messages
+  between Flutter and your desired platforms. For more
+  information, see [Call platform-specific code using platform channels](#example).
 
-* Kotlin or Java on Android
-* Swift or Objective-C on iOS
-* C++ on Windows
-* Objective-C on macOS
-* C on Linux
+* Use the `Pigeon` package to generate type-safe
+  platform-specific code. For more
+  information, see [Call platform-specific code using the Pigeon package](#pigeon).
 
-Flutter's builtin platform-specific API support
-doesn't rely on code generation,
-but rather on a flexible message passing style.
-Alternatively, you can use the [Pigeon][pigeon]
-package for [sending structured typesafe messages][]
-with code generation:
+Flutter supports the following platforms and
+platform-specific languages:
 
-* The Flutter portion of the app sends messages to its _host_,
-  the non-Dart portion of the app, over a platform channel.
-
-* The _host_ listens on the platform channel, and receives the message.
-  It then calls into any number of platform-specific APIs&mdash;using
-  the native programming language&mdash;and sends a response back to the
-  _client_, the Flutter portion of the app.
+* **Android**: Kotlin, Java
```

Comment:
This is just a question. Do we support Groovy as well?

Reply:
I'll double-check...

Reply:
From Reid... kind of in some places, but it's for build configuration. So nope, we are good!